evennia/contrib/turnbattle/tb_items.py

```diff
@@ -0,0 +1,1397 @@
+    
+    Args:
+        character (obj): Character to tick down the conditions of
+        turnchar (obj): Character whose turn it currently is
+        
+    Notes:
+        In combat, this is called on every fighter at the start of every character's turn. Out of
+        combat, it's instead called when a character's at_update() hook is called, which is every
+        30 seconds by default.
+    """
+    
+    for key in character.db.conditions:
+        # The first value is the remaining turns - the second value is whose turn to count down on.
+        condition_duration = character.db.conditions[key][0]
+        condition_turnchar = character.db.conditions[key][1]
+        # If the duration is 'True', then the condition doesn't tick down - it lasts indefinitely.
+        if not condition_duration is True:
+            # Count down if the given turn character matches the condition's turn character.
+            if condition_turnchar == turnchar:
+                character.db.conditions[key][0] -= 1
+            if character.db.conditions[key][0] <= 0:
+                # If the duration is brought down to 0, remove the condition and inform everyone.
+                character.location.msg_contents("%s no longer has the '%s' condition." % (str(character), str(key)))
+                del character.db.conditions[key]
+            
+def add_condition(character, turnchar, condition, duration):
+    """
+    Adds a condition to a fighter.
+    
+    Args:
+        character (obj): Character to give the condition to
+        turnchar (obj): Character whose turn to tick down the condition on in combat
+        condition (str): Name of the condition
+        duration (int or True): Number of turns the condition lasts, or True for indefinite
+    """
+    # The first value is the remaining turns - the second value is whose turn to count down on.
+    character.db.conditions.update({condition:[duration, turnchar]})
+    # Tell everyone!
+    character.location.msg_contents("%s gains the '%s' condition." % (character, condition))
+
+"""
+----------------------------------------------------------------------------
+CHARACTER TYPECLASS
+----------------------------------------------------------------------------
+"""
+
+
+class TBItemsCharacter(DefaultCharacter):
+    """
+    A character able to participate in turn-based combat. Has attributes for current
+    and maximum HP, and access to combat commands.
+    """
+
+    def at_object_creation(self):
+        """
+        Called once, when this object is first created. This is the
+        normal hook to overload for most object types.
+        """
+        self.db.max_hp = 100  # Set maximum HP to 100
+        self.db.hp = self.db.max_hp  # Set current HP to maximum
+        self.db.conditions = {} # Set empty dict for conditions
+        # Subscribe character to the ticker handler
+        tickerhandler.add(NONCOMBAT_TURN_TIME, self.at_update, idstring="update")
+        """
```

Comment:
This docstring belongs with the other docstring at the top of the method. You can make a `Notes:` section if you want to differentiate it from the header. Since docstrings are generally only used for methods/functions/classes, having it down here looks like a method went missing ... If you want a longer comment later in a method, use `#` comments instead.